packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -0,0 +1,212 @@
+import { pipeline } from 'stream';
+import { chain } from 'stream-chain';
+import {
+  IDestinationProvider,
+  ISourceProvider,
+  ITransferEngine,
+  ITransferEngineOptions,
+} from '../../types';
+
+export class TransferEngine implements ITransferEngine {
```

Comment:
Let's switch this to exporting an object or factory before merging

Reply:
Yes indeed